### PR TITLE
fix(3384): scenario outline parameter substitution

### LINKED ISSUE #3384: Parameter substitution from scenario outline works only for the first occurrence in a step
#### What are you trying to achieve?
Writing step as below

```
I type "<abc>" and "<pqr>" with "<abc>.test"
Examples : 
  | abc | pqr |   xyz    |
  |   I    | me | myself |
```

The resultant step should be:  
`I type I and me with I.test`

#### What do you get instead?

`I type I and me with <abc>.test`

> Provide console output if related. Use `--verbose` mode for more details.

```bash
# paste output here
```

> Provide test source code if related

```js
// paste test
```

### Details

* CodeceptJS version: 3.3.0
* NodeJS Version: 16.14.0
* Operating System: windows
* puppeteer || webdriverio || testcafe version (if related)
* Configuration file:

```js
# paste config here
```

### PR DESCRIPTION
**Pull request by snehabhandge**
## Motivation/Description of the PR
- Now the same column can be used multiple times in the same step in scenario outline 
- Resolves #3384.

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [ ] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [x] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/interfaces/gherkin.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ module.exports = (text, file) => {
             current[placeholder] = value;
             exampleSteps = exampleSteps.map((step) => {
               step = { ...step };
-              step.text = step.text.replace(`<${placeholder}>`, value);
+              step.text = step.text.split("<"+placeholder+">").join(value)
```
Comment (DavertMik) [inline lib/interfaces/gherkin.js:94]: wow, that's smart

**Comment by DavertMik** [comment]
thank you, never knew it can be used this way :open_mouth: 
thanks for the patch

**Comment by snehabhandge** [comment]
Hello @DavertMik, on which version of codeceptjs I can expect this fix?

**Comment by DavertMik** [comment]
3.3.5

working on it now

**Pull request merged**